src/test/java/com/znsio/rpap/WebExampleTest.java

```diff
@@ -0,0 +1,80 @@
+package com.znsio.rpap;
+
+import com.applitools.eyes.selenium.fluent.Target;
+import com.epam.reportportal.annotations.Step;
+import com.znsio.rpap.pages.WebExample;
+import com.znsio.rpap.utils.JsonDataManager;
+import com.znsio.rpap.utils.JsonDataProvider;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.lang.reflect.Method;
+
+import static com.znsio.rpi.utils.ReportPortalLogger.captureAndAttachScreenshot;
+
+
+public class WebExampleTest extends BaseTest {
+    private WebExample WebPage;
+
+    @Test(dataProvider = "getFromJson", priority = -1, description = "Validate Title of the Screen",
+            groups = {"visual"})
+    public void titleTest(String title) {
+
+        WebPage = page.getClassInstance(WebExample.class);
+        verifyPageTitle(title);
+    }
+
+    @Test(dataProvider = "getFromJson", description = "Validating login with valid username and password",
+            groups = {"visual"})
+    public void validLoginTest(String username, String password, String expectedMessage) throws InterruptedException {
+
+        performLogin(username, password);
+        verifyMessageAfterLogin(expectedMessage);
+    }
+
+    @Test(dataProvider = "getFromJson", description = "Validating login with invalid username and valid password",
+            groups = {"visual"})
+    public void invalidUserTest(String username, String password, String expectedMessage) throws InterruptedException {
+
+        performLogin(username, password);
+        verifyMessageAfterLogin(expectedMessage);
+    }
+
+    @Test(dataProvider = "getFromJson", description = "Validate login with valid username and invalid password",
+            groups = {"visual"})
+    public void invalidPasswordTest(String username, String password, String expectedMessage)
+            throws InterruptedException {
+
+        performLogin(username, password);
+        verifyMessageAfterLogin(expectedMessage);
+    }
+
+    @Step
+    private void verifyPageTitle(String title) {
```

Comment:
Move steps to another class. Shows a real-world example, instead of having everything in one class

Reply:
Moved to a seprate class `WebSteps`